### PR TITLE
5 Elixir tricks - Publish Tuesday, 8/15

### PR DESCRIPTION
**Pull request by Daniel-Xu**
[rendered](https://github.com/DockYard/reefpoints/pull/587/files?short_path=c423b2d#diff-c423b2d1ef2b6aa7733066fcb5b5e267)

### CODE REVIEW

File: source/posts/2017-07-25-elixir-tips.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Now you know `alias __MODULE__` just defines an alias for our Elixir module. This is very useful when used with `defstruct` which we will talk about next.
+
+In the following example, we pass `API.User` struct around to run some checks on our data. Instead of writing the full module name, we set up an alias `User` for it and pass that around. It's pretty concise and easy to read.
+
+![alias](https://i.imgur.com/3sl4Qmf.png)
+
+In case of module name changing, you can also do this:
+
+```
+alias __MODULE__, as: SomeOtherName
```
Comment (bcardarella) [inline source/posts/2017-07-25-elixir-tips.md]: this is interesting, but I am wondering if this is something we want to promote. Personally, I've relied upon `__MODULE__` within the module:

```elixir
def old?(%__MODULE__{name: name, age: age} = user)
```

I concede that readability of this kind of sucks

File: source/posts/2017-07-25-elixir-tips.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    %Game{status: status}
+  end
+  iex> Fun.Game.new(:won)
+  iex> %Fun.Game{status: :won, time: nil}
+```
+
+## `v()` function in `iex`
+
+Whenever I wrote a `GenServer` module, I usually want to start the server and check the result in `iex`.
+One thing that really bothers me is that I almost always forget to pattern matching the process pid, like this:
```
Comment (bcardarella) [inline source/posts/2017-07-25-elixir-tips.md]: "to pattern matching" -> "to pattern match"

File: source/posts/2017-07-25-elixir-tips.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+![genserver](https://i.imgur.com/DBLZzmZ.png)
+
+then, I need to type that command again with pattern matching:
+
+```
+{:ok, pid} = Metex.Worker.start_link
```
Comment (bcardarella) [inline source/posts/2017-07-25-elixir-tips.md]: can you add closing parens

**Comment by bcardarella** [comment]
Looks good, I had a few comments. I'd like to push back on the use of images for the code blocks however. The reason is that the images on imgur we don't have control over. So at some point in the future those images could be deleted and this post would be broken. Is there a good reason to not use the code blocks?

File: source/posts/2017-07-25-elixir-tips.md (hunk 1 above)
Comment (Daniel-Xu) [inline source/posts/2017-07-25-elixir-tips.md]: @bcardarella 
https://github.com/christopheradams/elixir_style_guide/issues/22#issuecomment-66783132

based on this thread, @chrismccord suggested to alias the module, but I'm not sure about his thought on `alias __MODULE__`.

**Comment by Daniel-Xu** [comment]
BTW, I replaced all the screenshot with code blocks.

File: source/posts/2017-07-25-elixir-tips.md (hunk 1 above)
Comment (bcardarella) [inline source/posts/2017-07-25-elixir-tips.md]: :+1:

**Pull request merged**